internal/cluster/cluster.go

```diff
@@ -1532,3 +1533,59 @@ func (m *Manager) updateSchedulableMastersForcedTrue(ctx context.Context, cluste
 
 	return nil
 }
+
+func (m *Manager) HandleVerifyVipsResponse(ctx context.Context, clusterID strfmt.UUID, stepReply string) error {
+	log := logutil.FromContext(ctx, m.log)
+	return m.db.Transaction(func(tx *gorm.DB) error {
+		var response models.VerifyVipsResponse
+		if err := json.Unmarshal([]byte(stepReply), &response); err != nil {
+			log.WithError(err).Error("HandleVerifyVipsResponse: unmarshal")
+		}
+		cluster, err := common.GetClusterFromDB(common.LoadTableFromDB(common.LoadTableFromDB(tx, common.APIVIPsTable), common.IngressVIPsTable),
+			clusterID, common.SkipEagerLoading)
+		if err != nil {
+			log.WithError(err).Errorf("HandleVerifyVipsResponse: getting cluster %s", clusterID.String())
+			return err
+		}
+		updated := false
+		for _, v := range response {
+			vipResponse := v
+			if vipResponse.Verification == nil {
+				continue
+			}
+			switch vipResponse.VipType {
+			case models.VipTypeAPI:
+				apiVip, _ := funk.Find(cluster.APIVips, func(apiVip *models.APIVip) bool { return apiVip.IP == vipResponse.Vip }).(*models.APIVip)
+				if apiVip != nil {
+					if apiVip.Verification == nil || *apiVip.Verification != *vipResponse.Verification {
+						apiVip.Verification = vipResponse.Verification
+						if err = tx.Save(apiVip).Error; err != nil {
+							log.WithError(err).Errorf("saving verification for api vip %s of cluster %s", apiVip.IP, clusterID.String())
+							return err
+						}
+						updated = true
+					}
+				}
```

Comment:
These lines are very similar to 1570-1580, is it difficult to create a generic function to cover both?
If so, at least worth extracting to two separate func, to make HandleVerifyVipsResponse a bit clearer:)

Reply:
The types models.APIVip and models.IngressVip are different so go does allow it.
It will need swagger change which I prefer not to do here.

Reply:
Yeah makes sense:/ So maybe just extract to a couple of functions if worth the effort. I mean just for reusability maybe and to make this func a bit clearer. Not crucial..